keyboards/ergodox_ez/keymaps/danielo515/keymap.c

```diff
@@ -0,0 +1,320 @@
+    KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_TRANSPARENT,
+    KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,
+
+    KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_UP         ,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_TRANSPARENT,KC_LEFT       ,KC_DOWN       ,KC_RIGHT      ,KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_TRANSPARENT,
+    KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT
+  ),
+
+  [8] = LAYOUT_ergodox(
+    KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_TRANSPARENT,KC_Q          ,KC_W          ,KC_E          ,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_ENTER      ,KC_A          ,KC_S          ,KC_D          ,KC_F          ,KC_TRANSPARENT,
+    KC_TRANSPARENT,KC_Z          ,KC_X          ,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_TRANSPARENT,
+    KC_SPACE      ,KC_LSHIFT     ,KC_TRANSPARENT,
+
+    TO(0)         ,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_PGDOWN     ,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_ENTER      ,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_TRANSPARENT,KC_TRANSPARENT,
+    KC_TRANSPARENT,
+    KC_TRANSPARENT,KC_TRANSPARENT,KC_TRANSPARENT
+  ),
+};
+// Used for blinking leds on layers higher than 7
+uint16_t blink_timer = 0;
+uint16_t blink_elapsed = 0;
+bool blink_led_1 = false;
+bool blink_led_2 = false;
+bool blink_led_3 = false;
+
+void matrix_scan_kb(void){
+    // first 7 layers do not need to blink
+    if(blink_led_1 || blink_led_2 || blink_led_3){
+        ergodox_board_led_off();
+        ergodox_right_led_1_off();
+        ergodox_right_led_2_off();
+        ergodox_right_led_3_off();
+        blink_elapsed = timer_elapsed(blink_timer);
+        if(blink_elapsed > 2000) { blink_timer = timer_read();}
+        if(blink_elapsed < 1000) {
+            if(blink_led_1) ergodox_right_led_1_on();
+            if(blink_led_2) ergodox_right_led_2_on();
+            if(blink_led_3) ergodox_right_led_3_on();
+        }
+    }
+    matrix_scan_user();
+};
+
+uint32_t layer_state_set_user(uint32_t state)
```

Comment:
This has been changed to a typedef, so it can be changed, to reduce the layer related processing.

```suggestion
layer_state_t layer_state_set_user(layer_state_t state)
```